### PR TITLE
noop: introduce logger that doesn't log anything

### PR DESCRIPTION
**Pull request by amery**

### CODE REVIEW

File: noop/noop.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+// WithEnabled passes the logger, but also indicates if it's enabled or not.
+// This logger is only enabled for Fatal entries
+func (nl *Logger) WithEnabled() (slog.Logger, bool) {
+	return nl, nl.Enabled()
+}
+
+// Print pretends to add a log entry with arguments handled in the manner of fmt.Print
+func (nl *Logger) Print(args ...any) {
+	if nl.Enabled() {
```
Comment (karasz) [inline noop/noop.go:37]: I would put his line in the print function level instead of all the functions that call print

**Review by karasz — CHANGES_REQUESTED** [review]

File: noop/noop.go (hunk 1 above)
Comment (amery) [reply noop/noop.go:37]: It was to avoid formatting strings unnecessarily

Comment (karasz) [reply noop/noop.go:37]: true, string formatting is quite costly

**Review by karasz — APPROVED** [review]
LGTM

**Pull request merged**